guardrails/validator_base.py

```diff
@@ -167,8 +171,36 @@ class Validator:
     _metadata = {}
 
     def __init__(
-        self, on_fail: Optional[Union[Callable, OnFailAction]] = None, **kwargs
+        self,
+        use_local: bool = None,
+        validation_endpoint: str = None,
+        on_fail: Optional[Union[Callable, OnFailAction]] = None,
+        **kwargs,
     ):
+        self.use_local = use_local
+        self.validation_endpoint = validation_endpoint
+        self.creds = Credentials.from_rc_file()
+
+        if not self.use_local:
+            if not self.creds:
+                raise PermissionError(
+                    "No credentials found! Please run 'guardrails configure' before"
+                    " making any validation requests."
+                )
+            self.hub_jwt_token = get_jwt_token(self.creds)
+            # If it wasn't set, fall back to credentials
+            if self.use_local is None:
```

Comment:
Isn't `set_local` checked at 184?  I don't see anything that changes it in the intervening 8 lines.  Or do we expect 184 to be False vs None?  If that's the case, 175 might need to be Optional[bool] to avoid the linter complaining.

Reply:
I reworked this. The weirdness comes from having the inputs override whatever is set in the config files on hub install. Should be a bit more understandable now